src/components/common/dashboard/OperationComponent.vue

```diff
@@ -0,0 +1,386 @@
+    import { useStore } from "@/use/store/store";
+    import { MutationTypes } from "@/use/store/mutation-types";
+    import { RejectionReasons } from "./reasons";
+    import { UC4Identifier } from "@/api/helpers/UC4Identifier";
+    import { showNotYetImplementedToast } from "@/use/helpers/Toasts";
+    import { Role } from "@/entities/Role";
+    import CertificateManagement from "@/api/CertificateManagement";
+    import GenericResponseHandler from "@/use/helpers/GenericResponseHandler";
+    import Router from "@/use/router";
+    import { approveMatriculation, rejectOperation } from "@/api/abstractions/FrontendSigning";
+    import { printOperation, printOperationTitle } from "@/api/abstractions/OperationPrinter";
+
+    export default {
+        name: "OperationComponent",
+        components: {},
+        props: {
+            operation: {
+                type: Object as () => Operation,
+                required: true,
+            },
+            enrollmentId: {
+                type: String,
+                required: true,
+            },
+            role: {
+                type: String,
+                required: true,
+            },
+            isArchive: {
+                type: Boolean,
+                required: false,
+                default: false,
+            },
+            watched: {
+                type: Boolean,
+                required: false,
+            },
+        },
+        emits: ["marked-read"],
+        setup(props: any, { emit }: any) {
+            const store = useStore();
+            const operation = ref(props.operation as Operation);
+            const approvals = operation.value.existingApprovals.users.length + operation.value.existingApprovals.groups.length;
+            const neededApprovals =
+                approvals + operation.value.missingApprovals.users.length + operation.value.missingApprovals.groups.length;
+            const missingApprovals = neededApprovals - approvals;
+
+            const isRejected = operation.value.state === OperationStatus.REJECTED;
+            const isPending = operation.value.state === OperationStatus.PENDING;
+            const isFinished = operation.value.state === OperationStatus.FINISHED || isRejected;
+            const actionRequired =
+                (operation.value.missingApprovals.users.includes(props.enrollmentID) ||
+                    operation.value.missingApprovals.groups.includes(props.role)) &&
+                !isRejected;
+
+            const sentApprove = ref(store.getters.treatedOperations.approved.includes(operation.value.operationId));
+            const sentReject = ref(store.getters.treatedOperations.rejected.includes(operation.value.operationId));
+            const provideReason = ref(false);
+            const selectedReason = ref("");
+            const writtenReason = ref("");
+            const isMyOperation = operation.value.initiator == props.enrollmentId;
```

Comment:
enrollmentId should/could be fetched here (it is cached so its fast)
keeps the props smaller

Reply:
changed in https://github.com/upb-uc4/ui-web/pull/785/commits/5e79a35183d277b197da266c8dc9e974868a2499